File: Server/SocialLogin/Models/Login/SignUpContent.cs
using System.Collections.Generic;

namespace SocialLogin.Models.Login
{
    public class SignUpContent
    {
        public List<CustomerOptInItem> CustomerOptInItems { get; set; }
        public List<CustomerConsentItem> CustomerConsentItems { get; set; }
        public List<string> SignInAuthenticators { get; set; }
    }

    public class CustomerOptInItem
    {
        public int Id { get; set; }
        public string Label { get; set; }
        public bool DefaultValue { get; set; }
    }

    public class CustomerConsentItem
    {
        public int Id { get; set; }
        public string Label { get; set; }
        public bool DefaultValue { get; set; }
    }
}

File: Client/src/models/initial/signUp/signUpContent.ts
import { Record, List } from 'immutable';
import { convertListToImmutable } from 'utils/immutable';
import { CustomerOptInItem, CustomerConsentItem } from './';

interface Interface {
    signInAuthenticators?: List<string>,
    customerOptInItems?: List<CustomerOptInItem>,
    customerConsentItems?: List<CustomerConsentItem>,
}

const initialValue = Record<Interface>({
    signInAuthenticators: undefined,
    customerOptInItems: undefined,
    customerConsentItems: undefined,
});

export class SignUpContent extends initialValue {
    constructor(args: any = {}) {
        super(Object.assign({}, args as Interface, {
            customerOptInItems: convertListToImmutable(args.customerOptInItems, CustomerOptInItem),
            customerConsentItems: convertListToImmutable(args.customerConsentItems, CustomerConsentItem),
        }));
    }
}
